src/ray/raylet/scheduling_resources.cc

```diff
@@ -9,39 +9,106 @@ namespace ray {
 
 namespace raylet {
 
-bool EqualsZeroEpsilon(double quantity) {
-  if ((quantity <= EPSILON) && (quantity >= -1 * EPSILON)) {
-    return true;
-  }
-  return false;
+FractionalResourceQuantity::FractionalResourceQuantity() {
+  resource_name_ = "";
+  resource_quantity_ = 0;
 }
 
-bool EqualsOneEpsilon(double quantity) {
-  if ((quantity <= 1 + EPSILON) && (quantity >= 1 - EPSILON)) {
-    return true;
-  }
-  return false;
+FractionalResourceQuantity::FractionalResourceQuantity(int resource_quantity) {
+  resource_name_ = "";
+  resource_quantity_ =
+      static_cast<int>(std::round(resource_quantity * conversion_factor_));
+}
+
+FractionalResourceQuantity::FractionalResourceQuantity(std::string resource_name,
+                                                       double resource_quantity) {
+  RAY_CHECK(resource_quantity > 0) << "Resource " << resource_name << " capacity is "
+                                   << resource_quantity
+                                   << ". Should have been greater than zero.";
+
+  resource_name_ = "";
+  resource_quantity_ =
+      static_cast<int>(std::round(resource_quantity * conversion_factor_));
```

Comment:
don't round, just truncate the fractional part

Reply:
The reason for this is that if you round, then you may end up with a situation where three tasks that each require 1/3 GPU can't get scheduled on a single GPU because they each require slightly more than 1/3.